src/histbuf.rs

```diff
@@ -1,72 +1,65 @@
-use generic_array::{sequence::GenericSequence, ArrayLength, GenericArray};
+use core::mem::MaybeUninit;
+use core::ptr;
+use core::slice;
 
 /// A "history buffer", similar to a write-only ring buffer of fixed length.
 ///
 /// This buffer keeps a fixed number of elements.  On write, the oldest element
 /// is overwritten. Thus, the buffer is useful to keep a history of values with
 /// some desired depth, and for example calculate a rolling average.
 ///
-/// The buffer is always fully initialized; depending on the constructor, the
-/// initial value is either the default value for the element type or a supplied
-/// initial value. This simplifies the API and is mostly irrelevant for the
-/// intended use case.
-///
 /// # Examples
 /// ```
 /// use heapless::HistoryBuffer;
-/// use heapless::consts::*;
 ///
 /// // Initialize a new buffer with 8 elements, all initially zero.
-/// let mut buf = HistoryBuffer::<_, U8>::new();
+/// let mut buf = HistoryBuffer::<_, 8>::new();
+///
+/// // Starts with no data
+/// assert_eq!(buf.recent(), None);
 ///
 /// buf.write(3);
 /// buf.write(5);
 /// buf.extend(&[4, 4]);
 ///
 /// // The most recent written element is a four.
-/// assert_eq!(buf.recent(), &4);
+/// assert_eq!(buf.recent(), Some(&4));
 ///
 /// // To access all elements in an unspecified order, use `as_slice()`.
 /// for el in buf.as_slice() { println!("{:?}", el); }
 ///
-/// // Now we can prepare an average of all values, which comes out to 2.
+/// // Now we can prepare an average of all values, which comes out to 4.
 /// let avg = buf.as_slice().iter().sum::<usize>() / buf.len();
-/// assert_eq!(avg, 2);
```

Comment:
Wow, this change should be explicitly noticed or even ported to a non const generic version.

Reply:
We will have it in the CHANGELOG as a breaking change